mmdet/models/anchor_heads/anchor_head.py

```diff
@@ -252,8 +302,10 @@ def get_bboxes_single(self,
             nms_pre = cfg.get('nms_pre', -1)
             if nms_pre > 0 and scores.shape[0] > nms_pre:
                 if self.use_sigmoid_cls:
+                    # The background class doesnt exist when using sigmoid
                     max_scores, _ = scores.max(dim=1)
                 else:
+                    # Remove background class scores when using softmax
                     max_scores, _ = scores[:, 1:].max(dim=1)
```

Comment:
For both sigmoid and softmax, this step is to get the maximum scores for all foreground classes. The comments may be a little confusing.